### DISCUSSION TITLE
Getting a 3-node k3s cluster going with Kairos 2.5 interactive?

### DISCUSSION THREAD

**Discussion by davidthewatson** [kairos-io/kairos#2097] [Q&A]
Hello...

I tried to get a 3-node k3s cluster going with Kairos 2.5 (the Ubuntu 23.04 build iso using k3s:
        
        kairos@pve:~$ k3s --version              
        k3s version v1.28.2+k3s1 (6330a5b4)
        go version go1.20.8
        kairos@pve:~$ kairos --version           
        version: 2.5.0, compiled with: go1.20.10
        kairos@pve:~$ uname -a
        Linux pve 6.2.0-39-generic #40-Ubuntu SMP PREEMPT_DYNAMIC Tue Nov 14 14:18:00 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

I had no trouble booting the iso via USB, selecting interactive and mesh network, finishing the install and rebooting without issue.

However, now that I've completed those steps, it seems that the system is not quite configured to run k3s as I have a cloud config generated in /oem/90_custom.yaml:
        
        #cloud-config
        
        install:
            device: /dev/nvme0n1
        k3s:
            enabled: true
        name: Config generated by the installer
        p2p:
            network_token: {redacted}
        stages:
            network:
                - users:
                    kairos:
                        groups:
                            - admin
                        name: kairos
                        passwd: {redacted}
                        ssh_authorized_keys:
                            - github:{redacted}

This looked un-registered to me so I tried to register it with `kairos register`; however, that command was not successful and I'm stuck. I did generate and submit a pull request to fix the skew between the doc's 99_custom.yaml and my version's 90_custom.yaml and I'm happy to contribute but I got the CI/CD checks on signoff prevented merging so I'll have to have another go at PR submission once I'm done yak-shaving the install/config. 

Any help to connect the dots between the 2.5 version and the gaps in the docs is sincerely appreciated. I'm happy to contribute code, doc, and help AMAP.

I love the look and feel of kairos and I'm super excited to use it as a replacement for my proxmox cluster which I've used for a decade launching greenfield startups from scratch.

Thanks,
David

**Comment by pgk831** [comment]
I also want a four machine cluster setup with Karios, but the master setup looks like a headache. There are no K3s, no Docker, and I can't find any documentation to step one forward in this case.
